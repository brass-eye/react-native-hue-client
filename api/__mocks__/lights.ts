const lights: any = {
	"4": {
		"state": {
			"on": true,
			"bri": 254,
			"hue": 7676,
			"sat": 199,
			"effect": "none",
			"xy": [
				0.5016,
				0.4151
			],
			"ct": 443,
			"alert": "none",
			"colormode": "xy",
			"mode": "homeautomation",
			"reachable": true
		},
		"swupdate": {
			"state": "noupdates",
			"lastinstall": null
		},
		"type": "Extended color light",
		"name": "Bedroom Lamp",
		"modelid": "LCT010",
		"manufacturername": "Philips",
		"productname": "Hue color lamp",
		"capabilities": {
			"certified": true,
			"control": {
				"mindimlevel": 1000,
				"maxlumen": 806,
				"colorgamuttype": "C",
				"colorgamut": [
					[
						0.6915,
						0.3083
					],
					[
						0.17,
						0.7
					],
					[
						0.1532,
						0.0475
					]
				],
				"ct": {
					"min": 153,
					"max": 500
				}
			},
			"streaming": {
				"renderer": true,
				"proxy": true
			}
		},
		"config": {
			"archetype": "sultanbulb",
			"function": "mixed",
			"direction": "omnidirectional"
		},
		"uniqueid": "00:17:88:01:02:35:23:25-0b",
		"swversion": "1.29.0_r21169",
		"swconfigid": "6A139B19",
		"productid": "Philips-LCT010-1-A19ECLv4"
	},
	"5": {
		"state": {
			"on": true,
			"bri": 254,
			"hue": 7676,
			"sat": 199,
			"effect": "none",
			"xy": [
				0.5016,
				0.4151
			],
			"ct": 443,
			"alert": "none",
			"colormode": "xy",
			"mode": "homeautomation",
			"reachable": true
		},
		"swupdate": {
			"state": "noupdates",
			"lastinstall": null
		},
		"type": "Extended color light",
		"name": "Bedroom Ceiling",
		"modelid": "LCT010",
		"manufacturername": "Philips",
		"productname": "Hue color lamp",
		"capabilities": {
			"certified": true,
			"control": {
				"mindimlevel": 1000,
				"maxlumen": 806,
				"colorgamuttype": "C",
				"colorgamut": [
					[
						0.6915,
						0.3083
					],
					[
						0.17,
						0.7
					],
					[
						0.1532,
						0.0475
					]
				],
				"ct": {
					"min": 153,
					"max": 500
				}
			},
			"streaming": {
				"renderer": true,
				"proxy": true
			}
		},
		"config": {
			"archetype": "sultanbulb",
			"function": "mixed",
			"direction": "omnidirectional"
		},
		"uniqueid": "00:17:88:01:02:33:29:37-0b",
		"swversion": "1.29.0_r21169",
		"swconfigid": "6A139B19",
		"productid": "Philips-LCT010-1-A19ECLv4"
	},
	"6": {
		"state": {
			"on": false,
			"bri": 124,
			"hue": 8418,
			"sat": 140,
			"effect": "none",
			"xy": [
				0.4573,
				0.41
			],
			"ct": 366,
			"alert": "none",
			"colormode": "ct",
			"mode": "homeautomation",
			"reachable": true
		},
		"swupdate": {
			"state": "noupdates",
			"lastinstall": null
		},
		"type": "Extended color light",
		"name": "Living room",
		"modelid": "LCT010",
		"manufacturername": "Philips",
		"productname": "Hue color lamp",
		"capabilities": {
			"certified": true,
			"control": {
				"mindimlevel": 1000,
				"maxlumen": 806,
				"colorgamuttype": "C",
				"colorgamut": [
					[
						0.6915,
						0.3083
					],
					[
						0.17,
						0.7
					],
					[
						0.1532,
						0.0475
					]
				],
				"ct": {
					"min": 153,
					"max": 500
				}
			},
			"streaming": {
				"renderer": true,
				"proxy": true
			}
		},
		"config": {
			"archetype": "sultanbulb",
			"function": "mixed",
			"direction": "omnidirectional"
		},
		"uniqueid": "00:17:88:01:02:35:22:c4-0b",
		"swversion": "1.29.0_r21169",
		"swconfigid": "6A139B19",
		"productid": "Philips-LCT010-1-A19ECLv4"
	}
}
export default lights;