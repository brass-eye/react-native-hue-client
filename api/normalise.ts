import { Lights } from "../hue/light";

export const normaliseLights = (res: any): Lights => {
    return Object.keys(res).map((lightId) => ({
        lightId: lightId,
        type: res[lightId].type,
        name: res[lightId].name,
        on: res[lightId].state.on,
        brightness: res[lightId].state.bri,
        hue: res[lightId].state.hue,
        saturation: res[lightId].state.sat,
        reachable: res[lightId].state.reachable
    }))
}
