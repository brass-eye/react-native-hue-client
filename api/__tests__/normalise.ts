import lights from '../__mocks__/lights'
import { normaliseLights } from '../normalise';

describe('testing changing shape of api response', () => {
    it('transforms lights', () => {
        expect(normaliseLights(lights)).toMatchSnapshot();
    });
}) 