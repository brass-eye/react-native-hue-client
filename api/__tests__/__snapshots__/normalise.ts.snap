// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing changing shape of api response transforms lights 1`] = `
Array [
  Object {
    "brightness": 254,
    "hue": 7676,
    "lightId": "4",
    "name": "Bedroom Lamp",
    "on": true,
    "reachable": true,
    "saturation": 199,
    "type": "Extended color light",
  },
  Object {
    "brightness": 254,
    "hue": 7676,
    "lightId": "5",
    "name": "Bedroom Ceiling",
    "on": true,
    "reachable": true,
    "saturation": 199,
    "type": "Extended color light",
  },
  Object {
    "brightness": 124,
    "hue": 8418,
    "lightId": "6",
    "name": "Living room",
    "on": false,
    "reachable": true,
    "saturation": 140,
    "type": "Extended color light",
  },
]
`;
